src/Symfony/Component/Dotenv/Dotenv.php

```diff
@@ -60,21 +60,31 @@ public function load($path/*, ...$paths*/)
     /**
      * Sets values as environment variables (via putenv, $_ENV, and $_SERVER).
      *
-     * Note that existing environment variables are never overridden.
+     * Existing environment variables are never overridden, unless they are listed in the SYMFONY_DOTENV_VARS env var.
      *
      * @param array $values An array of env variables
      */
     public function populate($values)
     {
+        $loadedVars = array_flip(explode(',', getenv('SYMFONY_DOTENV_VARS')));
+        unset($loadedVars['']);
+
         foreach ($values as $name => $value) {
-            if (isset($_ENV[$name]) || isset($_SERVER[$name]) || false !== getenv($name)) {
+            if (!isset($loadedVars[$name]) && (isset($_ENV[$name]) || isset($_SERVER[$name]) || false !== getenv($name))) {
                 continue;
             }
 
             putenv("$name=$value");
             $_ENV[$name] = $value;
             $_SERVER[$name] = $value;
+
+            $loadedVars[$name] = true;
         }
+
+        $loadedVars = implode(',', array_keys($loadedVars));
```

Comment:
@nicolas-grekas, I think you're right. I've moved that block inside an `if` statement.